File: rank.py
from pymongo import MongoClient
import time
client = MongoClient()
db = client.paper_rank
collection_from = db.papers
collection_to = db.rank
cnt = 0
rets = []
tic = time.time()
for p in collection_from.find():
	p_rank = {}
	for k in p["loss_value"].keys():
		lv = p["loss_value"][k]
		rank = collection_from.find({"loss_value." + k: {"$gt": lv}}).count() + 1
		# print len(list(collection_from.find({"loss_value." + k: {"$gt": lv}}, {"loss_value." + k: 1, "_id":0})))
		# collection_from.find({"loss_value." + k: {"$gt": lv}})
		# collection_from.aggregate([
		# 	{"$match": {"loss_value." + k: {"$gt": lv}}},
		# 	{"$group": { "_id": "null", "count": {"$sum": 1}}}
		# ])
		p_rank[k] = rank
	# collection_from.update_one({"_id": p["_id"]}, {"$set": {"rank": p_rank}})
	rets.append({"_id": p["_id"], "rank": p_rank})
	if cnt%2 == 0:
		collection_to.insert_many(rets)
		rets = []
		print cnt
		print time.time() - tic
	cnt += 1
	if cnt == 6:
		break

File: rank.js
var client = new Mongo();
var db = client.getDB("paper_rank");
var collection_from = db.papers
var collection_to = db.rank
var cnt = 0;
var rets = [];
var ps = collection_from.find();
// var tic = new Date().getTime();
while(ps.hasNext()){
	var p = ps.next();
	var p_rank = {};
	var keys = Object.keys(p.loss_value);
	for (var k_index in keys){
		var key = keys[k_index];
		var lv = p["loss_value"][key];
		var field = "loss_value." + key;
		rank = collection_from.find({[field]: {"$gt": lv}}).count() + 1;
		p_rank[key] = rank;
	}
	rets.push({"_id": p["_id"], "rank": p_rank});
	if (cnt%200 === 0){
		collection_to.insertMany(rets);
		rets = [];
		// var toc = new Date().getTime();
		// print(toc - tic);
		print(cnt);
	}
	cnt += 1;
	// if (cnt===6){
	// 	break;
	// }
}